cmake does not check if gtest is installed

The cmake is not checking if gtest is installed in the system and fails during the compilation with missing  headers:

Fields2Cover/tests/cpp/headland_generator/const_hl_test.cpp:7:10: fatal error: gtest/gtest.h: Nie ma takiego pliku ani katalogu
    7 | #include <gtest/gtest.h>                                                                                                                                                                           
      |          ^~~~~~~~~~~~~~~

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,6 +86,7 @@ FetchContent_MakeAvailable(matplotlib_cpp)
 
 find_package(GDAL 3.0 REQUIRED)
 
+find_package(GTest REQUIRED)
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
 
 set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,7 +86,6 @@ FetchContent_MakeAvailable(matplotlib_cpp)
 
 find_package(GDAL 3.0 REQUIRED)
 
-find_package(GTest REQUIRED)
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
 
 set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
@@ -255,6 +254,7 @@ endif()
 #####################################################
 
 if (BUILD_TESTS)
+  find_package(GTest REQUIRED)
   include(CTest)
   enable_testing()
   add_custom_target(check COMMAND GTEST_COLOR=1 ${CMAKE_CTEST_COMMAND}

